bundle/edu.gemini.catalog/src/main/scala/edu/gemini/catalog/api/CatalogQuery.scala

```diff
@@ -41,30 +49,87 @@ sealed trait CatalogQuery {
   }
 }
 
+trait CatalogQueryWithRange extends CatalogQuery { this: CatalogQuery =>
+  val magnitudeRange: MagnitudeRange
```

Comment:
As an aside, I think I'd rather not use an abstract `val` here.  If you declare this as `def magnitudeRange: MagnitudeRange` you can still implement it in the extending class with a `val` and it avoids having it temporarily initialized to `null`.